app/src/main/java/com/example/weather/StreamUtils.java

```diff
@@ -0,0 +1,41 @@
+package com.example.weather;
+
+import android.util.Log;
+
+import androidx.annotation.Nullable;
+
+import java.io.BufferedReader;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+public class StreamUtils {
+    private static final String TAG = Thread.currentThread().getStackTrace()[2].getClassName();
```

Comment:
```suggestion
    private static final String TAG = StreamUtils.class.getSimpleName();
```

Why Thread.currentThread().getStackTrace()[2].getClassName()?